web/src/typing_data.ts

```diff
@@ -64,6 +65,25 @@ export function get_topic_typists(stream_id: number, topic: string): number[] {
     return muted_users.filter_muted_user_ids(typists);
 }
 
+export function add_edit_message_typing_id(message_id: number): void {
+    if (!edit_message_typing_ids.includes(message_id)) {
+        edit_message_typing_ids.push(message_id);
+    }
+}
+
+export function remove_edit_message_typing_id(message_id: number): boolean {
+    if (!edit_message_typing_ids.includes(message_id)) {
+        return false;
+    }
+    const message_id_index = edit_message_typing_ids.indexOf(message_id);
+    edit_message_typing_ids.splice(message_id_index, 1);
+    return true;
```

Comment:
Whenever you find yourself writing messy splicing logic like this, you should stop yourself as it's a sign you've got the wrong data structure.

Reply:
ok right... I'll use a Set in that case.